cmd/prometheus/main.go

```diff
@@ -259,6 +261,9 @@ func main() {
 	a.Flag("storage.remote.read-max-bytes-in-frame", "Maximum number of bytes in a single frame for streaming remote read response types before marshalling. Note that client might have limit on frame size as well. 1MB as recommended by protobuf by default.").
 		Default("1048576").IntVar(&cfg.web.RemoteReadBytesInFrame)
 
+	a.Flag("storage.exemplars.exemplars-limit", "Maximum number of exemplars to store in in-memory exemplar storage total.").
+		Default("100000").IntVar(&cfg.ExemplarsLimit)
```

Comment:
Can we use `uint` or check for negative value? currently a negative value panics:

```
panic: runtime error: makeslice: len out of range

goroutine 288 [running]:
github.com/prometheus/prometheus/tsdb.NewCircularExemplarStorage(...)
        /home/roidelapluie/dev/prometheus/tsdb/exemplar.go:74
github.com/prometheus/prometheus/tsdb.NewHead(0x3321540, 0xc000148c80, 0x32e4900, 0xc00084da40, 0xc000914a20, 0xffffffffffffffff, 0x6ddd00, 0x2ada88d, 0x5, 0x32fe200, ...)
        /home/roidelapluie/dev/prometheus/tsdb/head.go:317 +0x9c
github.com/prometheus/prometheus/tsdb.open(0x2ada88d, 0x5, 0x32e4900, 0xc00084da40, 0x3321540, 0xc000148c80, 0xc00027ed20, 0xc00005ab40, 0x3, 0xa, ...)
        /home/roidelapluie/dev/prometheus/tsdb/db.go:656 +0x5fa
github.com/prometheus/prometheus/tsdb.Open(0x2ada88d, 0x5, 0x32e4900, 0xc00084da40, 0x3321540, 0xc000148c80, 0xc00027ed20, 0xc000e8de48, 0x7f7682, 0xc0009b22c0)
        /home/roidelapluie/dev/prometheus/tsdb/db.go:533 +0xbc
main.openDBWithMetrics(0x2ada88d, 0x5, 0x32e4900, 0xc0009af260, 0x3321540, 0xc000148c80, 0xc00027ed20, 0x0, 0x0, 0x0)
        /home/roidelapluie/dev/prometheus/cmd/prometheus/main.go:875 +0x10d
main.main.func20(0x0, 0x0)
        /home/roidelapluie/dev/prometheus/cmd/prometheus/main.go:787 +0x1ff
github.com/oklog/run.(*Group).Run.func1(0xc00027ed80, 0xc0009b3780, 0xc00084d980)
        /home/roidelapluie/dev/prometheus/vendor/github.com/oklog/run/group.go:38 +0x27
created by github.com/oklog/run.(*Group).Run
        /home/roidelapluie/dev/prometheus/vendor/github.com/oklog/run/group.go:37 +0xbb

```

Reply:
I think you meant to comment on the other PR? But yes, I will make a change there, I think checking for negative when creating the exemplar storage is the way to go.

Reply:
Oh there are 2 PR's :(